### ISSUE TITLE
-Werror means project doesn't compile with gcc 5.4.0

### ISSUE THREAD

**Issue by dhasenan** [InfiniteAxis/SmaugFUSS#13]
Errors produced:

```
comm.c: In function ‘void nanny_get_name(DESCRIPTOR_DATA*, char*)’:
comm.c:1878:44: warning: comparison of constant ‘255’ with boolean expression is always false [-Wbool-compare]
    if( check_playing( d, argument, FALSE ) == BERR )
                                            ^
comm.c:1950:12: warning: comparison of constant ‘255’ with boolean expression is always false [-Wbool-compare]
    if( chk == BERR )
            ^
comm.c: In function ‘void nanny_get_old_password(DESCRIPTOR_DATA*, char*)’:
comm.c:2043:12: warning: comparison of constant ‘255’ with boolean expression is always false [-Wbool-compare]
    if( chk == BERR )
            ^
comm.c: In function ‘void do_name(CHAR_DATA*, const char*)’:
comm.c:3414:51: warning: comparison of constant ‘255’ with boolean expression is always false [-Wbool-compare]
    if( check_playing( ch->desc, argument, FALSE ) == BERR )
                                                   ^
skills.c: In function ‘bool check_tumble(CHAR_DATA*, CHAR_DATA*)’:
skills.c:4674:66: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
    if( !IS_NPC( victim ) && !victim->pcdata->learned[gsn_tumble] > 0 )
                                                                  ^
```

Since compiler warnings change without warning, consider removing `-Werror` from the CC flags in the makefile.

**Comment by hatetank** [comment]
`if( !IS_NPC( victim ) && !(victim->pcdata->learned[gsn_tumble] > 0) )`

Above takes care of the logical-not-parentheses warning.

Removing -Werror is not a good idea, but as a temporary workaround to unsorting that BERR constant (and/or check_playing being a big fat liar on its return type) screwing up comm.c compilation, just add -Wno-error=bool-compare to W_FLAGS

Ideally somebody would submit a proper fix for that, but just hacking around at it by changing return types winds up introducing some side effects I don't care to understand at the moment.

**Comment by Arthmoor** [comment]
This was fixed at some point for the current 1.9.x codebase.
